spark/src/main/scala/io/delta/sharing/spark/DeltaSharingDataSource.scala

```diff
@@ -37,13 +40,37 @@ private[sharing] class DeltaSharingDataSource extends RelationProvider with Data
     val path = parameters.getOrElse("path", throw new IllegalArgumentException(
       "'path' is not specified. If you use SQL to create a Delta Sharing table, " +
         "LOCATION must be specified"))
+
+    var cdfOptions: mutable.Map[String, String] = mutable.Map.empty
+    val caseInsensitiveParams = new CaseInsensitiveStringMap(parameters.asJava)
+    if (DeltaSharingDataSource.isCDFRead(caseInsensitiveParams)) {
+      cdfOptions = mutable.Map[String, String](DeltaSharingDataSource.CDF_ENABLED_KEY -> "true")
+      if (caseInsensitiveParams.containsKey(DeltaSharingDataSource.CDF_START_VERSION_KEY)) {
+        cdfOptions(DeltaSharingDataSource.CDF_START_VERSION_KEY) = caseInsensitiveParams.get(
+          DeltaSharingDataSource.CDF_START_VERSION_KEY)
+      }
+      if (caseInsensitiveParams.containsKey(DeltaSharingDataSource.CDF_START_TIMESTAMP_KEY)) {
+        cdfOptions(DeltaSharingDataSource.CDF_START_TIMESTAMP_KEY) = caseInsensitiveParams.get(
+          DeltaSharingDataSource.CDF_START_TIMESTAMP_KEY)
+      }
+      if (caseInsensitiveParams.containsKey(DeltaSharingDataSource.CDF_END_VERSION_KEY)) {
+        cdfOptions(DeltaSharingDataSource.CDF_END_VERSION_KEY) = caseInsensitiveParams.get(
+          DeltaSharingDataSource.CDF_END_VERSION_KEY)
+      }
+      if (caseInsensitiveParams.containsKey(DeltaSharingDataSource.CDF_END_TIMESTAMP_KEY)) {
+        cdfOptions(DeltaSharingDataSource.CDF_END_TIMESTAMP_KEY) = caseInsensitiveParams.get(
+          DeltaSharingDataSource.CDF_END_TIMESTAMP_KEY)
+      }
```

Comment:
Do we need to validate these, or is it expected that they are already validated?

Reply:
do you mean the unit test? There's no unit test for this class only, I will add integration test once all code are in place.

Reply:
No I meant validating the options (for example, only one of startVersion/startTimestamp is set).

Reply:
I see. I didn't add it because I was thinking of leaving this check to the server. Should we check it on the connector as well? Maybe we can skip?

Reply:
Ok, let the server do the checks.